rama-http-backend/src/client/proxy/layer/proxy_connector/service.rs

```diff
@@ -50,8 +52,26 @@ impl<S: Clone> Clone for HttpProxyConnector<S> {
 
 impl<S> HttpProxyConnector<S> {
     /// Creates a new [`HttpProxyConnector`].
+    ///
+    /// Protocol version is set to HTTP/1.1 by default.
     pub(super) fn new(inner: S, required: bool) -> Self {
-        Self { inner, required }
+        Self {
+            inner,
+            required,
+            version: Some(Version::HTTP_11),
+        }
+    }
+
+    /// Set the HTTP version to use for the CONNECT request.
+    pub(super) fn set_version(&mut self, version: Version) -> &mut Self {
```

Comment:
Please also add the `with_` versions of these 2: `with_version(mut self, ...` and `with_auto_version(mut self)`